pkg/utils/number.go

```diff
@@ -15,13 +15,22 @@ limitations under the License.
 */
 package utils
 
+import "time"
+
 func Min(x, y int) int {
 	if x < y {
 		return x
 	}
 	return y
 }
 
+func MinDuration(x, y time.Duration) time.Duration {
```

Comment:
optional: I think go's generics might be useful here:
```suggestion
func Min[T constraints.Ordered](x, y T) T {
```